package.json

```diff
@@ -170,6 +170,7 @@
     "serve-favicon": "~2.5.0",
     "speakingurl": "~14.0.1",
     "stathat": "0.0.8",
+    "styled-components": "~4.4.1",
```

Comment:
I wonder why styled-components as opposed to other react styling libraries. I took a quick look, and seems a lot of them have been deprecated over time, so styled-components is looking good from that perspective.

The only other one that caught my eye from a brief look is something based on CSS-modules (e.g. https://github.com/gajus/babel-plugin-react-css-modules).

I don't have a strong opinion about these things, really, although maybe others do! I can see advantages to having style defined without the file, or in seperate files.

Reply:
The choice these days [seems to be](https://css-tricks.com/the-fragmented-but-evolving-state-of-css-in-js/) mostly between [styled-components](https://github.com/styled-components/styled-components) (the most popular) or [Emotion](https://github.com/emotion-js/emotion). They have cross-compatible API so migrating from one to another [is fairly simple](https://github.com/WordPress/gutenberg/pull/17963#issuecomment-552842261).

I don't think the differences in our use would be significant; Emotion might be nicer when publishing components to NPM as it can isolate styles more effectively and it's also a little more performant than Styled components. Emotion can be themed easily (not sure about styled-components) and Styled components plays nicely with React Native (not sure about Emotion). There might be some gotchas in each library with server-side-rendering from w hat I've heard. None of these are our concerns, anyway.

CSS modules are really nice too but if you prefer something like this, probably better to clean-cut and go all in CSS-in-JS. :-) 

_That said_, just importing style files directly from JS already improves the developer experience significantly, together with more solid CSS selector naming conventions. Would that still be enough?

There's going to be a performance penalty with CSS-in-JS — which might be ok if dev experience improves, but just something to keep in mind.

[Facebook announced](https://www.youtube-nocookie.com/embed/9JZHodNR184) they're working on a solution that's supposedly solving performance issues; we might want to move to that eventually. Even migrating to it from any other CSS-in-JS library should be smooth enough, since APIs across different frameworks are typically quite similar. I'm sure someone will even write Codemods to help to migrate automatically.

Reply:
For completeness: We discussed this on Slack.

Consensus among people involved in the discussion seemed to be to keep the `styled-components` library in experimental mode and see if we like it. If we don't like it we will remove it in the future. Or we will replace it with the fancy, soon-to-come new proposals for css-in-js mentioned above.